### PR TITLE
Fixes docs mismatched with implementation

### PR DESCRIPTION
**Pull request by yue9944882**
**What this PR does / why we need it**:

Some document is far different with the implementation, which is really confusing.

Manually fixed them and appended more information.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/service/service_controller.go [hunk 1]
```diff
@@ -235,9 +235,8 @@ func (s *ServiceController) init() error {
 	return nil
 }
 
-// Returns an error if processing the service update failed, along with a time.Duration
-// indicating whether processing should be retried; zero means no-retry; otherwise
-// we should retry in that Duration.
+// processServiceUpdate writes the actual state of incoming service to the cache and operates loadbalancers
```
Comment (MrHohn) [inline pkg/controller/service/service_controller.go]: Might not be necessary to mention the cache. FYI this cache is being removed by https://github.com/kubernetes/kubernetes/pull/54569.

Comment (yue9944882) [reply pkg/controller/service/service_controller.go]: @MrHohn Reasonable.

**Comment by yue9944882** [comment]
/assign @MrHohn 

PTAL Thx :)

**Review by MrHohn — APPROVED** [review]
Thanks!
/lgtm

**Pull request merged**